File: src/test/java/org/seekers/python/TestPythonInstaller.java
package org.seekers.python;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.nio.file.Files;
import java.nio.file.attribute.PosixFilePermissions;

class TestPythonInstaller {

    static PythonSettings settings = new PythonSettings();
    static PythonInstaller installer = new PythonInstaller(settings);

    @Test
    void download() {
        Assertions.assertDoesNotThrow(() -> installer.download());
    }

    @Test
    void workflow() {
        Assertions.assertDoesNotThrow(() ->  installer.workflow());
    }

    @Test
    void permissions() {
        Assertions.assertDoesNotThrow(() -> Files.setPosixFilePermissions(new File(settings.getProperty("folder"),
                "client/run_client").toPath(), PosixFilePermissions.fromString("rwxrwxrwx")));
    }

    @Test
    void clean() {
        Assertions.assertDoesNotThrow(() -> installer.clean());
    }

    @Test
    void clear() {
        Assertions.assertDoesNotThrow(() -> installer.clear());
    }
}

File: src/test/kotlin/org/seekers/python/TestPythonInstaller.kt
package org.seekers.python

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.attribute.PosixFilePermissions

internal class TestPythonInstaller {

    @Test
    fun workflow() {
        Assertions.assertDoesNotThrow { installer.workflow() }
    }

    @Test
    fun download() {
        Assertions.assertDoesNotThrow { installer.download() }
    }

    @Test
    fun permissions() {
        Assertions.assertDoesNotThrow<Path> {
            Files.setPosixFilePermissions(File(settings["folder"], "client/run_client")
                .toPath(), PosixFilePermissions.fromString("rwxrwxrwx")) }
    }

    @Test
    fun clean() {
        Assertions.assertDoesNotThrow { installer.clean() }
    }

    @Test
    fun clear() {
        Assertions.assertDoesNotThrow { installer.clear() }
    }

    companion object {
        var settings: PythonSettings = PythonSettings()
        var installer: PythonInstaller = PythonInstaller(settings)
    }
}
